### PR TITLE
FIX Prevent segfaults in CLI runner

### LINKED ISSUE #2964: TypeError: getWasmTableEntry(...) is not a function
## 🐛 Bug

```
pyodide.asm.js:10 Pyodide has suffered a fatal error. Please report this to the Pyodide maintainers.
pyodide.asm.js:10 The cause of the fatal error was:
pyodide.asm.js:10 TypeError: getWasmTableEntry(...) is not a function
    at invoke_ii (pyodide.asm.js:10:1544237)
    at 00bcba5a:0x54516
    at invoke_viiiiiiiiiiiiiii (pyodide.asm.js:10:1548668)
    at 00bcba5a:0x414a2
    at invoke_viii (pyodide.asm.js:10:1545014)
    at 00bcba5a:0x56286
    at method_call_trampoline (pyodide.asm.js:10:216572)
    at pyodide.asm.wasm:0x162162
    at method_call_trampoline (pyodide.asm.js:10:216572)
    at pyodide.asm.wasm:0x126317
API.fatal_error @ pyodide.asm.js:10
Module.callPyObjectKwargs @ pyodide.asm.js:10
Module.callPyObject @ pyodide.asm.js:10
apply @ pyodide.asm.js:10
apply @ pyodide.asm.js:10
runPython @ pyodide.asm.js:10
main @ (index):15
await in main (async)
(anonymous) @ (index):22
pyodide.asm.js:10 Stack (most recent call first):
pyodide.asm.js:10   File "<exec>", line 5 in <module>
pyodide.asm.js:10   File "/lib/python3.10/site-packages/_pyodide/_base.py", line 304 in run
pyodide.asm.js:10   File "/lib/python3.10/site-packages/_pyodide/_base.py", line 435 in eval_code
(index):21 Uncaught (in promise) TypeError: getWasmTableEntry(...) is not a function
    at invoke_ii (pyodide.asm.js:10:1544237)
    at 00bcba5a:0x54516
    at invoke_viiiiiiiiiiiiiii (pyodide.asm.js:10:1548668)
    at 00bcba5a:0x414a2
    at invoke_viii (pyodide.asm.js:10:1545014)
    at 00bcba5a:0x56286
    at method_call_trampoline (pyodide.asm.js:10:216572)
    at pyodide.asm.wasm:0x162162
    at method_call_trampoline (pyodide.asm.js:10:216572)
    at pyodide.asm.wasm:0x126317
```

### To Reproduce

https://github.com/kitao/pyxel/tree/maturin

* First install the maturin beta version: `pip install -U --pre maturin`
* Build with maturin: `RUSTUP_TOOLCHAIN=nightly maturin build -o dist --release --target wasm32-unknown-emscripten`
* Run `./scripts/start_server` and open `http://localhost:8000` in browser, see the error in console.

### Expected behavior

No `TypeError: getWasmTableEntry(...) is not a function` in console.

### Environment

- Pyodide Version: 0.21.0
- Browser version: 104.0.5112.79
- Any other relevant information:

### PR DESCRIPTION
**Pull request by hoodmane**
After many hours of debugging, I minimized the problem down to this: https://github.com/emscripten-core/emscripten/issues/22052

Thanks to @henryiii for reporting. See thread here for my comments while I was debugging:
https://github.com/scikit-hep/boost-histogram/pull/938

Resolves #2964.

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by hoodmane** [comment]
Actually, we could just backport https://github.com/emscripten-core/emscripten/pull/22053

File: src/js/load-package.ts [hunk 1]
```diff
@@ -555,6 +555,9 @@ export async function loadPackage(
       Array.from(toLoad.values()).map(({ installPromise }) => installPromise),
     );
 
+    // Warning: this sounds like it might not do anything important, but it
+    // fills in the GOT. There can be segfaults if we leave it out.
+    // See https://github.com/emscripten-core/emscripten/issues/22052
     Module.reportUndefinedSymbols();
```
Comment (ryanking13) [inline src/js/load-package.ts:561]: Huh, If I remember correctly, that function throws if it encounters a symbol that doesn't actually exist, but I may be misremembering something.

**Review by ryanking13 — COMMENTED** [review]
Thanks for analysing and fixing the issue @hoodmane!

File: src/js/load-package.ts (hunk 1 above)
Comment (hoodmane) [reply src/js/load-package.ts:561]: It fills in the Global Object Table:
```
    var reportUndefinedSymbols = () => {
      for (var [symName, entry] of Object.entries(GOT)) {
        if (entry.value == 0) {
          var value = resolveGlobalSymbol(symName, true).sym;
          if (!value && !entry.required) {
            // Ignore undefined symbols that are imported as weak.
            continue;
          }
          if (typeof value == "function") {
            /** @suppress {checkTypes} */
            entry.value = addFunction(value, value.sig);
          } else if (typeof value == "number") {
            entry.value = value;
          } else {
            throw new Error(
              `bad export type for '${symName}': ${typeof value}`,
            );
          }
        }
      }
    };
```

**Review by ryanking13 — APPROVED** [review]
Thanks, let's see if your upstream PR will get merged.

**Comment by hoodmane** [comment]
I think upstream PR is going to take a while and probably change, but I wouldn't be against using the patch in that PR here instead of this change.

**Comment by ryanking13** [comment]
> I think upstream PR is going to take a while and probably change, but I wouldn't be against using the patch in that PR here instead of this change.

Sure, I have no objection with it.

**Comment by hoodmane** [comment]
Except that it seems to break pyxel and pygame?? I'll have to investigate further.

**Comment by hoodmane** [comment]
The problem has to do with the linking of JavaScript symbols.

**Pull request merged**